docs/how_to_contribute/math/how_to_add_prime_fields.md

```diff
@@ -8,7 +8,21 @@ _Note_: We have our own development conventions. Please read the [conventions do
 
 Begin by creating a directory named `<new_prime_field>_prime` in `/tachyon/math/finite_field/`. Add a `BUILD.bazel` file into this directory. Note that once parameters are added to `BUILD.bazel`, Bazel will automatically generate the prime field code based on these parameters when it builds the target.
 
-Choose among `generate_prime_fields()`, `generate_fft_prime_fields()`, or `generate_large_fft_prime_fields()` for code generation, depending on the prime field type. For more information, refer to [`prime_field_generator`](/tachyon/math/finite_fields/generator/prime_field_generator/build_defs.bzl).
+### [Prime field generator]((/tachyon/math/finite_fields/generator/prime_field_generator/build_defs.bzl))
+
+Choose the prime field generator depending on the prime field type.
+
+1. **`generate_prime_fields()`**: The basic prime field generator requires the following properties.
+    - `name`: The name of the prime field.
+    - `namespace`: Set the namespace, commonly set as `tachyon::math`.
+    - `class_name`: The name of the class, usually the same as the name of the prime field (in PascalCase).
+    - `modulus`: The modulus value of the prime field.
+    - `flag`: The name of the flag exposed in the prime field's configuration (`kIs<PrimeFieldName>`).
+2. **`generate_fft_prime_fields()`**: In addition to the properties of `generate_prime_fields`, `generate_fft_prime_fields()` requires a `subgroup_generator`. subgroup generator is a value used to generate elements of a subgroup within the prime field, facilitating Fast Fourier Transform (FFT) operation.
```

Comment:
```suggestion
2. **`generate_fft_prime_fields()`**: In addition to the properties of `generate_prime_fields`, `generate_fft_prime_fields()` requires a `subgroup_generator`. The `subgroup_generator` is a value used to generate elements of a subgroup within the prime field, facilitating Fast Fourier Transform (FFT) operations.

Reply:
If anything, it may make more sense to put the second sentence below in the category of `subgroup_generator`. Following the convention above, it should be inserted as "The value used to generate elements of a subgroup within the prime field, facilitating Fast Fourier Transform (FFT) operations."